### PR TITLE
[RHOAIENG-6641] Backport 2.9 SSRF fixes to 2.8.x

### PR DESCRIPTION
**Pull request by alexcreasy**
Closes: [RHOAIENG-6641](https://issues.redhat.com//browse/RHOAIENG-6641)

## Description

* Removes proxy API from backend and backports service API from 2.9
* Updates Pipelines and TrustyAI features to use new API
* Updates cypress and unit tests

## How Has This Been Tested?

Run on a cluster deployed with ODH 2.8.x zstream release.

* Verify Pipelines features are not broken
* Verify Bias features are not broken
* Smoke test the rest of the dashboard to ensure no breakages

## Test Impact

* Existing tests updated to work with back ported APIs
* 
## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: backend/src/routes/api/service/pipelines/index.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    kind: 'DataSciencepipelinesApplication',
+    plural: 'datasciencepipelinesapplications',
+  },
+  {
+    port: 8888,
+    prefix: 'ds-pipeline-',
+  },
+  {
+    // Use port forwarding for local development:
+    // kubectl port-forward -n <namespace> svc/ds-pipeline-pipelines-definition 8888:8888
```
Comment (christianvogt) [inline backend/src/routes/api/service/pipelines/index.ts:17]: Head of code uses port `8443` why is this different?

File: frontend/package.json [hunk 2]
```diff
@@ -37,8 +37,8 @@
     "storybook": "storybook dev -p 6006 -s ../config",
     "cypress:open": "cypress open --project src/__tests__/cypress",
     "cypress:run": "cypress run -b chrome --project src/__tests__/cypress",
-    "cypress:open:mock": "MOCK=1 WS_PORT=9002 npm run cypress:open",
-    "cypress:run:mock": "MOCK=1 WS_PORT=9002 npm run cypress:run",
+    "cypress:open:mock": "MOCK=1 WS_PORT=9002 npm run cypress:open --",
+    "cypress:run:mock": "MOCK=1 WS_PORT=9002 npm run cypress:run --",
```
Comment (christianvogt) [inline frontend/package.json:41]: This change seems unrelated.

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineDeleteRuns.cy.ts [hunk 3]
```diff
@@ -187,55 +181,37 @@ describe('Pipeline runs', () => {
...
-          data: {},
-        });
-      });
-
       cy.wait('@getRuns').then(() => {
         pipelineRunJobTable.findEmptyState().should('exist');
       });
     });
-    it('Test delete a single run from triggered', () => {
+    it.only('Test delete a single run from triggered', () => {
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineDeleteRuns.cy.ts:214]: Undo this change.

File: frontend/src/concepts/trustyai/context/TrustyAIContext.tsx [hunk 4]
```diff
@@ -72,23 +63,23 @@ export const TrustyAIContextProvider: React.FC<TrustyAIContextProviderProps> = (
   const contextValue = React.useMemo(
     () => ({
       namespace,
-      hasCR: !!explainabilityNamespaceCR,
+      hasCR: !!trustyNamespaceCR,
       crInitializing: !crLoaded,
       serverTimedOut,
       ignoreTimedOut,
-      serviceLoadError,
+      crLoadError,
```
Comment (christianvogt) [inline frontend/src/concepts/trustyai/context/TrustyAIContext.tsx:70]: Well this is interesting. There's no type error wrt the context prop type `TrustyAIContextProps`. Nor is there an issue in `head` on `main` branch. The type context type still includes `serviceLoadError`.

File: backend/src/routes/api/service/pipelines/index.ts (hunk 1 above)
Comment (alexcreasy) [reply backend/src/routes/api/service/pipelines/index.ts:17]: Because the DSPA created for v1 pipelines doesn't support TLS, this is the only working port exposed by the service on ODH 2.8.x.

File: frontend/package.json (hunk 2 above)
Comment (alexcreasy) [reply frontend/package.json:41]: Ahh yeah, I brought that in to make my life easier... I'll remove it.

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineDeleteRuns.cy.ts (hunk 3 above)
Comment (alexcreasy) [reply frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineDeleteRuns.cy.ts:214]: Nice spot thanks

File: backend/src/routes/api/service/pipelines/index.ts (hunk 1 above)
Comment (alexcreasy) [reply backend/src/routes/api/service/pipelines/index.ts:17]: I was just about to post a screenshot of this... it's possible I'm going mad here - there _is_ an oath port exposed on the service at port 8443?

I'll change this now.

File: frontend/src/concepts/trustyai/context/TrustyAIContext.tsx (hunk 4 above)
Comment (alexcreasy) [reply frontend/src/concepts/trustyai/context/TrustyAIContext.tsx:70]: Yeah this is weird... even if I explicitly type the memo it doesn't complain hmmm. Seems TS isn't picking up on the indirection here. I tested this by assigning the props inline to the value prop and suddenly it's complaining. If you inline the memo TS stops complaining again....

Comment (alexcreasy) [reply frontend/src/concepts/trustyai/context/TrustyAIContext.tsx:70]: Bug for main: https://issues.redhat.com/browse/RHOAIENG-7251

**Comment by christianvogt** [comment]
Tested on a cluster setup by Alex. Ensured that both pipelines and trusty requests went to the new `/api/service/*` paths. Tested various pipeline actions including uploading a file.

/lgtm
